src/preprocessing/passes/sygus_inference.cpp

```diff
@@ -259,44 +307,19 @@ bool SygusInference::simplify(std::vector<Node>& assertions)
     if (itffv != ff_var_to_ff.end())
     {
       Node ff = itffv->second;
-      Expr body = it->second;
-      std::vector<Expr> args;
-      // if it is a non-constant function
-      if (lambda.getKind() == LAMBDA)
-      {
-        for (const Node& v : lambda[0])
-        {
-          args.push_back(v.toExpr());
-        }
-        body = it->second[1];
-      }
-      Trace("sygus-infer") << "Define " << ff << " as " << it->second
-                           << std::endl;
-      final_ff.push_back(ff);
-      final_ff_sol.push_back(it->second);
-      master_smte->defineFunction(ff.toExpr(), args, body);
+      Node body = Node::fromExpr(it->second);
+      Trace("sygus-infer") << "Define " << ff << " as " << body << std::endl;
+      funs.push_back(ff);
+      sols.push_back(body);
     }
-  }
-
-  // apply substitution to everything, should result in SAT
-  for (unsigned i = 0, size = assertions.size(); i < size; i++)
-  {
-    Node prev = assertions[i];
-    Node curr = assertions[i].substitute(final_ff.begin(),
-                                         final_ff.end(),
-                                         final_ff_sol.begin(),
-                                         final_ff_sol.end());
-    if (curr != prev)
+    else
     {
-      curr = Rewriter::rewrite(curr);
-      Trace("sygus-infer-debug")
-          << "...rewrote " << prev << " to " << curr << std::endl;
-      assertions[i] = curr;
+      Assert(false);
```

Comment:
You don't need this or the `else` since you already have an assertion for getting into the `if`